src/classes/RD2_EntryFormController.cls

```diff
@@ -294,7 +294,9 @@ public with sharing class RD2_EntryFormController {
             npe03__Recurring_Donation__c rd =  (npe03__Recurring_Donation__c) JSON.deserialize(
                 jsonRecord, npe03__Recurring_Donation__c.class
             );
-            UTIL_Http.Response response = commitmentService.handleCommitment(rd, null, paymentMethodToken);
+
+            npe03__Recurring_Donation__c oldRd = getOldRecurringDonation(rd.Id);
```

Comment:
Pass Contact and Account Id from front-end into this method instead of querying oldRd.